disnake/guild.py

```diff
@@ -4150,3 +4156,116 @@ async def timeout(
 
         data = await self._state.http.edit_member(self.id, user.id, reason=reason, **payload)
         return Member(data=data, guild=self, state=self._state)
+
+    # TODO: naming
+    async def automod_rules(self) -> List[AutomodRule]:
+        """|coro|
+
+        Retrieves the guild's auto moderation rules.
+
+        Requires the :attr:`~Permissions.manage_guild` permission.
+
+        .. versionadded:: 2.6
+
+        Raises
+        ------
+        Forbidden
+            You do not have proper permissions to retrieve auto moderation rules.
+        HTTPException
+            Retrieving the rules failed.
+
+        Returns
+        -------
+        List[:class:`AutomodRule`]
+            The guild's auto moderation rules.
+        """
+        data = await self._state.http.get_auto_moderation_rules(self.id)
+        return [AutomodRule(data=rule_data, guild=self) for rule_data in data]
+
+    async def create_automod_rule(
+        self,
+        name: str,
+        *,
+        # TODO: should this default to the only existing type currently?
+        event_type: AutomodEventType,
```

Comment:
> typing for the other types 

Not sure what you mean, could you elaborate?

Reply:
Hm, while I did misread the code initially, my point stands, albiet on the wrong argument. I meant trigger_type. We could add overloads for the metadata based on the trigger type.

Reply:
Opted not to do this, as it would be annoying and a typing pain.

Reply:
I'd rather keep these only in the documentation and not enforce them through overloads, since it would unnecessarily restrict calls with a non-literal `AutoModTriggerType` (consider a command that has a parameter of that type).